### ISSUE TITLE
question about e107::lan()

### ISSUE THREAD

**Issue by Jimmi08** [e107inc/e107#3985]
I have used the variable in the language file. After using e107::lan()  is not available anymore

It's alphabet, so it is language relative, but it's an array.

When I use normal include_once(), it works.  I could use the string and create the array from it after loading lan file, but could somebody explain to me what is the advantage of using e107::lan() or e107::includeLan()? 

I mean - during debugging I noticed the attempt to load so many not existed or not used files that it lost its effectivity for me.  Why load lan files from other plugins when I don't need them?  If I want them loaded in my plugin, I would load them myself. 

Thanks

**Comment by Moc** [comment]
> I have used the variable in the language file. 

What do you mean exactly, can you give an example?

**Comment by Jimmi08** [comment]
@Moc I am porting some script for e107. This script is old but updated for php7.2 (it works as separate installation, so there are not php issues).  I am trying to move some its part as e107 plugin so they could work together (not only with alt_auth plugin).  Line by line, in fact, to learn how it works (it has some features what e107 can only dream now)

So f.e. in file authors.php they include file header.php and inside it they include language file with variable $alphabet. No OOP stuff.

`$alphabet = array( 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', _OTHER );`

it's accessible later without using word global (of course, except functions)

I can't do it in e107.

At first I put e107::lan() to e_header.php  Then I got that maybe because of addon stuff (object) this will not work.

Then I put it directly to file with e107::lan().  It worked only if I added global $alfabeth at the beginning of the lan file. 

but when I just used only include file (without e107 stuff), it worked without using word global.  

This is caused probably my PHP limited knowledge, but it just looks weird that so easy thing is so complicated now. 

PS. move to e107 templates, shortcodes (simple parsing for now) and rendering (tablerender) is so easy to use and it works, I am just replacing that old code.
